### PR TITLE
feat: adds delta support to gcode viewer

### LINKED ISSUE #519: Gcode live view not working for circular bed (delta printer)
### What happened

Gcode live view is empty when using on detla printer but it is working good on rectangular bed. I guess it is because of round bed have origin at 0;0
image

### What did you expect to happen

Gcode live view is visible when using round bed

### How to reproduce

Start any print on delta printer and load current file to gcode live view

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by susisstrolch**
Fixes #519

Signed-off-by: Juergen Orschiedt <juorschiedt@gmail.com>

### CODE REVIEW

File: src/components/widgets/gcode-preview/GcodePreview.vue [hunk 1]
```diff
@@ -211,6 +228,14 @@ export default class GcodePreview extends Mixins(StateMixin) {
 
   panning = false
 
+  get isDelta (): boolean {
+    return this.$store.getters['printer/getPrinterSettings']('printer.kinematics') === 'delta'
```
Comment (pedrolamas) [inline src/components/widgets/gcode-preview/GcodePreview.vue:232]: I believe your changes should work with both `delta` and `rotary_delta`, so can you make sure you check for both?

**Review by pedrolamas — CHANGES_REQUESTED** [review]

**Comment by susisstrolch** [comment]
Even I've added a test for "rotary_delta" I'm not sure it will work out.
I simply don't know how to compute the bedsize for a rotary.
So I'd suggest to omit the "rotary_delta" for the moment until we get a clear description how the bedsize (radius) is computed.

**Comment by pedrolamas** [comment]
> So I'd suggest to omit the "rotary_delta" for the moment until we get a clear description how the bedsize (radius) is computed.

That is fair and makes sense! Let us then wait for someone with such a printer to do that bit!

**Review by pedrolamas — DISMISSED** [review]
LGTM

**Comment by susisstrolch** [comment]
The current implementation doesn't hurt so much, because print radius will be preset with 100.0 for rotary_delta.
However we could add an additional printer_radius setting for those who want to have an exact picture.

**Comment by pedrolamas** [comment]
> The current implementation doesn't hurt so much, because print radius will be preset with 100.0 for rotary_delta. However we could add an additional printer_radius setting for those who want to have an exact picture.

My personal knowledge on how a delta works is quite limited so I can't evaluate exactly what this involves.

If this works with your delta, I'm happy to merge the changes, as code-wise, this looks perfect to me and I've regression tested this to make sure everything is fine!

**Comment by susisstrolch** [comment]
Ok, so let's go on and let's wait for the rotary_delta guys...

File: src/components/widgets/gcode-preview/GcodePreview.vue [hunk 2]
```diff
@@ -271,6 +297,9 @@ export default class GcodePreview extends Mixins(StateMixin) {
       this.flipY ? -(y.max - y.min) : 0
     ]
 
+    if (this.isDelta) {
```
Comment (matmen) [inline src/components/widgets/gcode-preview/GcodePreview.vue:300]: This branch could be moved before `const transform = ...` to save a memory allocation for an unused reference

**Review by matmen — APPROVED** [review]
LGTM overall, just one tiny thing I would change ;)

**Review by pedrolamas — APPROVED** [review]

**Review by matmen — APPROVED** [review]
LGTM, just waiting for a delta user to chime in..

**Pull request merged**